app/services/course_search_service.rb

```diff
@@ -51,44 +52,64 @@ def call
 
 private
 
-  def sites_with_distance_from_origin
+  def distance_with_university_area_adjustment
+    university_provider_type = Provider.provider_types[:university]
+    university_location_area_radius = 10
+    <<~EOSQL.gsub(/^[\s\t]*/, "").gsub(/[\s\t]*\n/, " ").strip
+      (CASE
+        WHEN provider.provider_type = '#{university_provider_type}'
+          THEN (distance - #{university_location_area_radius})
+        ELSE distance
+      END) as boosted_distance
+    EOSQL
+  end
+
+  def locatable_sites
     site_status = SiteStatus.arel_table
     sites = Site.arel_table
 
-    # Create virtual table with sites and site statuses
-    sites_with_status = site_status.join(sites).on(site_status[:site_id].eq(sites[:id]))
-
-    # only want new and running sites
-    new_and_running_sites = sites_with_status.where(site_status[:status].in(%w[new_status running]))
+    # Only running and published site statuses
+    running_and_published_criteria = site_status[:status].eq(SiteStatus.statuses[:running]).and(site_status[:publish].eq(SiteStatus.publishes[:published]))
 
     # we only want sites that have been geocoded
-    geocoded_new_and_running_sites = new_and_running_sites.where(sites[:latitude].not_eq(nil).and(sites[:longitude].not_eq(nil)))
+    has_been_geocoded_criteria = sites[:latitude].not_eq(nil).and(sites[:longitude].not_eq(nil))
 
     # only sites that have a locatable address
     # there are some sites with no address1 or postcode that cannot be
     # accurately geocoded. We don't want to return these as the closest site.
     # This should be removed once the data is fixed
-    locatable_new_and_running_sites = geocoded_new_and_running_sites.where(sites[:address1].not_eq("").or(sites[:postcode].not_eq("")))
+    locatable_address_criteria = sites[:address1].not_eq("").or(sites[:postcode].not_eq(""))
+
+    # Create virtual table with sites and site statuses
+    locatable_sites = site_status.join(sites).on(site_status[:site_id].eq(sites[:id]))
+
+    locatable_sites = locatable_sites.where(running_and_published_criteria)
+    locatable_sites = locatable_sites.where(has_been_geocoded_criteria)
+    locatable_sites.where(locatable_address_criteria)
```

Comment:
Not to be petty, but could we make this a little bit more Ruby-like:

```ruby
    locatable_sites = site_status.join(sites).on(site_status[:site_id].eq(sites[:id]))
                        .where(running_and_published_criteria)
                        .where(has_been_geocoded_criteria)
                        .where(locatable_address_criteria)
```

Or even, since we've already given the name of the result in the method:

```ruby
    site_status.join(sites).on(site_status[:site_id].eq(sites[:id]))
      .where(running_and_published_criteria)
      .where(has_been_geocoded_criteria)
      .where(locatable_address_criteria)
```

Reply:
sure I just borrowed the existing style...